ffi/diplomat/src/bidi.rs

```diff
@@ -34,9 +34,7 @@ pub mod ffi {
         /// Creates a new [`ICU4XBidi`] from locale data.
         #[diplomat::rust_link(icu::properties::bidi::BidiClassAdapter::new, FnInStruct)]
         pub fn try_new(provider: &ICU4XDataProvider) -> DiplomatResult<Box<ICU4XBidi>, ICU4XError> {
-            use icu_provider::serde::AsDeserializingBufferProvider;
-            let provider = provider.0.as_deserializing();
-            maps::load_bidi_class(&provider)
+            maps::load_bidi_class(&provider.0)
```

Comment:
thought: for future-proofing it might be worth having a `provider.get_provider()` method so we can change the internal representaiton without churn

but also we're not going to be doing that that much so I don't think we _need_ to, just an idea

Reply:
`provider.0` points to the inner enum, which we can change without churn, so long as whatever we put at `provider.0` implements the traits

Reply:
okay, that works